### ISSUE TITLE
octave 4.2-rc02 does not support gui

### ISSUE THREAD

**Issue by sunilshah** [Homebrew/homebrew-science#4418]
### Indicate troubleshooting steps taken
Ran brew update
Ran brew upgrade octave

### Bug reports / build errors:

==> Installing homebrew/science/octave
==> Downloading https://homebrew.bintray.com/bottles-science/octave-4.2.0-rc2.el_capitan.bottle.1.tar.gz
######################################################################## 100.0%
==> Pouring octave-4.2.0-rc2.el_capitan.bottle.1.tar.gz
Warning: homebrew/science/octave dependency gcc was built with a different C++ standard
library (libstdc++ from clang). This may cause problems at runtime.
==> Caveats

Octave's graphical user interface is disabled; compile Octave with the option
--with-gui to enable it.

Several graphics toolkit are available. You can select them by using the command
'graphics_toolkit' in Octave.  Individual Gnuplot terminals can be chosen by setting
the environment variable GNUTERM and building gnuplot with the following options:

  setenv('GNUTERM','qt')    # Requires QT; install gnuplot --with-qt5
  setenv('GNUTERM','x11')   # Requires XQuartz; install gnuplot --with-x11
  setenv('GNUTERM','wxt')   # Requires wxmac; install gnuplot --with-wxmac
  setenv('GNUTERM','aqua')  # Requires AquaTerm; install gnuplot --with-aquaterm

You may also set this variable from within Octave. For printing the cairo backend
is recommended, i.e., install gnuplot with --with-cairo, and use

  print -dpdfcairo figure.pdf

When using the native Qt or fltk toolkits then invisible figures do not work because
osmesa is incompatible with Mac's OpenGL. The usage of gnuplot is recommended.

#### Brief summary of issue:

How do I compile octave with gui?  

Can I issue 

brew install octave --with-gui ?  

Or do I need to issue

brew install octave -s --with-gui?

Lastly, has anyone installed octave 4.2-rc2 on Sierra?

**Comment by dpo** [comment]
`brew install octave --with-gui` will build from source. Please close this issue if it works for you.

**Comment by schoeps** [comment]
Actually it should be
```
brew install octave --with-qt5
```
but `--with-gui` is still supported (we should change the text of the caveat). However, it will not compile with recent Xcode >=8. That is why we disabled the GUI.

> Lastly, has anyone installed octave 4.2-rc2 on Sierra?

Yes but it's currently painful. You must use a patched qt 5.7.0 (https://github.com/Homebrew/homebrew-core/pull/5619) and a patched qscintilla2 (https://github.com/sqlitebrowser/sqlitebrowser/issues/776). Both formulas will be hopefully released soon.

**Comment by sunilshah** [comment]
Thanks.  I will wait for the patched formulas to be released for 10.12.  Once they work, I will close the issue.

By the way, upgrading Mac OS 10.11 to OS 10.12 makes installed octave 4.0.3  to fail at run time.  I will wait for the patched formulas before upgrading other Macs  to OS 10.12.

**Comment by sunilshah** [comment]
This issue below is not about octave-gui but about getting octave-cli to work correctly on Mac OS 10.12 for octave 4.2-rc2

1) 
brew update
brew install octave 

runs to completion but plot command  (gnuplot) does not work.  
setenv("GNUTERM","qt"), or "x11", or "aqua" does not help.

gnuplot is always called with aqua from octave.

2)
brew install octave --with-qt5   

also runs to completion but  has the same behavior as in 1) above.

--without-qt5 also has the same issue. 

3) Re-installing gnuplot with various options meets the same issue.

4) Installing aquaterm with 

brew install Caskroom/cask/aquaterm
brew install gnuplot --with-aquaterm

allows plot in octave-cli to work correctly. 

I wonder if this behavior has anything to do with qt5.7 and qscintilla2 patches.  

Schoeps, let me know if I need to open a new issue.

**Comment by schoeps** [comment]
> setenv("GNUTERM","qt"), or "x11", or "aqua" 
This requires gnuplot to be installed with the corresponding options `--with-...`. What switches did you choose? 

In either case `brew install gnuplot --with-qt5` and `brew install octave --with-qt5` are still broken.

**Comment by sunilshah** [comment]
In case 1) I did not specify any flags to the brew install octave command.

In case 3) I did test --with-qt5 --without-qt5  --with-aqua

**Comment by schoeps** [comment]
On Sierra neither `gnuplot` nor `octave` compile `--with-qt5`. You can use `gnuplot` with x11 or aquaterm backend or fltk.
```
brew uninstall octave
brew uninstall gnuplot
brew install Caskroom/cask/aquaterm
brew install gnuplot --with-aquaterm  --with-x11 --with-cairo
brew install octave
```
Then the following options should work within octave
```
>graphics_toolkit('gnuplot');
>setenv('GNUTERM','x11');
>figure;
>plot(1:10,1:10);
```
or 
```
>graphics_toolkit('gnuplot');
>setenv('GNUTERM','aqua');
>figure;
>plot(1:10,1:10);
```
and since the fltk backend is compiled by default:
```
>graphics_toolkit('fltk');
>figure;
>plot(1:10,1:10);
```

**Comment by sunilshah** [comment]
This is really helpful -thanks Schoeps. 

Yes, each of the above worked on Sierra.  

Additionally, with that build I also verified that the following worked as well.
1)  octave kernel in Jupyter with inline plots 
2) pkg install -forge control signal

**Comment by dpo** [comment]
Should be fixed by #4475.
